lib/main.js

```diff
@@ -6,19 +6,22 @@ import logger from './logger'; // logger needs to remain first of imports
 import _ from 'lodash';
 import { server as baseServer, routeConfiguringFunction as makeRouter } from 'appium-base-driver';
 import { asyncify } from 'asyncbox';
-import { default as getParser, getDefaultArgs } from './parser';
+import { default as getParser } from './parser';
+import { getDefaultServerArgs } from './server-parser';
 import { logger as logFactory, util } from 'appium-support';
 import {
   showConfig, checkNodeOk, validateServerArgs,
   warnNodeDeprecations, validateTmpDir, getNonDefaultArgs,
-  getDeprecatedArgs, getGitRev, APPIUM_VER
+  getGitRev, APPIUM_VER
 } from './config';
+import DriverConfig from './driver-config';
+import { runDriverCommand } from './driver-cli';
 import { AppiumDriver } from './appium';
 import registerNode from './grid-register';
 import { inspectObject } from './utils';
 
 
-async function preflightChecks (parser, args, throwInsteadOfExit = false) {
+async function preflightChecks (parser, args, driverConfig, throwInsteadOfExit = false) {
```

Comment:
it looks suspicious if a method needs more than 3 arguments, it, probably, needs to be refactored or the arguments should be combined